sonarr/episode.go

```diff
@@ -34,19 +34,54 @@ type Episode struct {
 	Series                   *Series        `json:"series"`
 }
 
-// GetSeriesEpisodes returns all episodes for a series by series ID.
+// GetEpisode represents the input parameters for an episode api request.
+type GetEpisode struct {
+	// Set seriesID to get episodes for a specific series. Set to zero to get all episodes.
+	SeriesID int64
+	// Set seasonNumber to get episodes for a specific season. Set to zero to get all episodes.
+	SeasonNumber int
+	// Set episodeIds to get episodes for a specific set of ID's. Set to zero to get all episodes.
+	EpisodeIDs []int64
+	// Set episodeFileId to get episodes for a specific file. Set to zero to get all episodes.
+	EpisodeFileID int64
+	// Set includeImages to include images for each episode.
+	IncludeImages bool
+}
+
+// GetSeriesEpisodes returns all episodes for a series by Series ID, Season Number, Episode ID's, or EpisodeFileID.
 // You can get series IDs from GetAllSeries() and GetSeries().
-func (s *Sonarr) GetSeriesEpisodes(seriesID int64) ([]*Episode, error) {
-	return s.GetSeriesEpisodesContext(context.Background(), seriesID)
+func (s *Sonarr) GetSeriesEpisodes(getEpisode *GetEpisode) ([]*Episode, error) {
+	return s.GetSeriesEpisodesContext(context.Background(), getEpisode)
 }
 
-// GetSeriesEpisodesContext returns all episodes for a series by series ID.
+// GetSeriesEpisodesContext returns all episodes for a series by Series ID, Season Number, Episode ID's, or EpisodeFileID.
 // You can get series IDs from GetAllSeries() and GetSeries().
-func (s *Sonarr) GetSeriesEpisodesContext(ctx context.Context, seriesID int64) ([]*Episode, error) {
+func (s *Sonarr) GetSeriesEpisodesContext(ctx context.Context, getEpisode *GetEpisode) ([]*Episode, error) {
 	var output []*Episode
 
-	req := starr.Request{URI: bpEpisode, Query: make(url.Values)}
-	req.Query.Add("seriesId", fmt.Sprint(seriesID))
+	params := make(url.Values)
+
+	if getEpisode.SeriesID > 0 {
+		params.Set("seriesId", starr.Itoa(getEpisode.SeriesID))
+	}
+
+	if getEpisode.SeasonNumber > 0 {
+		params.Set("seasonNumber", starr.Itoa(int64(getEpisode.SeasonNumber)))
```

Comment:
instead of `starr.Itoa(int64())`
use `strconv.Itoa()`

Reply:
SeasonNumber is an int and not an int64

Reply:
oh haha I just noticed you put strconv